CircuitPlaygroundFirmata/CircuitPlaygroundFirmata.ino

```diff
@@ -1130,18 +1134,28 @@ void setup()
   // Serial1.begin(57600);
   // Firmata.begin(Serial1);
   // then comment out or remove lines 701 - 704 below
-  Firmata.begin(57600);
+  SerialW.begin(57600);
+  Serial.begin(57600);
+
+  // Listen for either serial port type to connect.
+  while (!SerialW && !Serial) {
+    #if defined(DEMO_MODE)
+      runDemo();   // this will 'demo' the board off, so you know its working, until the serial port is opened
+    #endif
+  }
+
+  if (SerialW) {
+    Firmata.begin(SerialW);
+  }
+  if (Serial) {
+    Firmata.begin(Serial);
+  }
 
   // Tell Firmata to ignore pins that are used by the Circuit Playground hardware.
   // This MUST be called or else Firmata will 'clobber' pins like the SPI CS!
   pinConfig[28] = PIN_MODE_IGNORE;   // Pin 28 = D8 = LIS3DH CS
   pinConfig[26] = PIN_MODE_IGNORE;   // Messes with CS too?
 
-#if defined(DEMO_MODE)
```

Comment:
please lave this in, you can have DEMO_MODE undefined

Reply:
I switched the `if defined` into the `while (!SerialW && !Serial)` and moved it up to line 1140 so that by the time we `Firmata.begin` we've connected to either SerialW or Serial. Let me know if that works for you! Thanks!